src/common/transformations/src/transformations/common_optimizations/remove_multi_subgraph_op_dangling_params.cpp

```diff
@@ -159,47 +157,50 @@ bool ov::pass::RemoveMultiSubGraphOpDanglingParamsResults::run_on_model(const st
                         }
                     }
                 }
-                for (auto input_index : required_inputs_indices) {
-                    if (input_index > removed_loop_idx) {
-                        new_required_inputs_indices.insert(input_index - 1);
-                    } else {
-                        new_required_inputs_indices.insert(input_index);
-                    }
+            };
+
+            const auto update_required_input_indicies = [](std::set<uint64_t>& required_inputs_indices,
+                                                           uint64_t removed_input_idx) {
+                std::set<uint64_t> new_required_inputs_indices;
```

Comment:
I know it's quite time consuming to create tests for such transformations
but could you please cover the fixed case?

Reply:
Current tests cover it e.g. `RemoveIfDanglingParametersFromBodiesAndInputs` but the issue is using `operator[]` was access some undefined data and it was luck just it worked (the MSVC debug build and test run should detect access out of bound).

Current modification add checks for removed inputs and using `operator[]` is safe.
Removing checks:
`!util::contains(body_removed_indices, current_body_parameter_idx)`
`!util::contains(op_removed_inputs, current_input_idx)`
There will be like test failures (seg fault) .

I can think about additional test (which fail with previous implementation), but during refactoring the current test failed when I've done something wrong.

Reply:
Test added.